.github/workflows/build_macos.yml

```diff
@@ -0,0 +1,52 @@
+name: Build (macOS)
+
+on: [push, pull_request]
+
+jobs:
+  macOS:
+    runs-on: macOS-latest
+    strategy:
+      matrix:
+        configuration: ["Debug", "Release"]
+    steps:
+    - name: Define Build Options
+      shell: python
+      run: |
+        if "${{ matrix.configuration }}" == "Debug":
+          print('Configuring for Debug')
+          print('::set-env name=BUILD_PARAM::')
+          print('::set-env name=ARTIFACT_NAME::xqemu-debug')
+        else:
+          print('Configuring for Release')
+          print('::set-env name=BUILD_PARAM::--release')
+          print('::set-env name=ARTIFACT_NAME::xqemu-release')
+    - name: Clone Tree
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - name: Install Dependencies
+      run: |
+        brew update
+        brew install \
+          glib \
+          pixman \
+          sdl2 \
+          libepoxy \
+          coreutils \
+          pkg-config \
+          ccache
+    - name: Initialize ccache
+      id: cache
+      uses: actions/cache@v1
+      with:
+        path: /tmp/ccache
+        key: cache-${{ runner.os }}-${{ matrix.configuration }}-${{ github.sha }}
+        restore-keys: cache-${{ runner.os }}-${{ matrix.configuration }}-
```

Comment:
I'm not sure how any of this really works, but I'm a bit surprised to see this end in a dash symbol `-` (which implies more data follows, as for the `key` property, which is actually followed by the hash).

Reply:
This is to work around what is an oversight of [actions/cache](https://github.com/actions/cache) (see link for docs). The problem is that if there's a cache hit, the cache will *not* be updated after the build--which means we only get objects from the first time the cache is created.

Unfortunately, there's no way to force an update of the cache post-build. However, I found by correlating the cache key with the commit hash causing the event (which will always be a cache-miss), then providing a restore-key common *prefix* (which actions/cache will use as a prefix in a reverse-chronological search of the cache list), I can get the cache to (1) always get the latest cached objects and (2) always create an updated cache. Naturally, old cache entries will be evicted as time goes on.

AS for the terminating `-`, this is the style used in GitHub docs.

Reply:
I'd have to read up on this. So I'll just blindly trust your expertise for this PR.
Here's some naive thoughts that I still had.

> Unfortunately, there's no way to force an update of the cache post-build. However, I found *[hacky-logic]*

I feel like this should be an issue on actions/cache (about how to force this update; and wether *[hacky logic]* is a suitable / future-proof workaround).
I'm a bit worried that this behaviour will break.

> [...] then providing a restore-key common prefix (which actions/cache will use as a prefix in a reverse-chronological search of the cache list),
> As for the terminating -, this is the style used in GitHub docs.

I'm surprised to hear that this is a prefix.
The README of actions/cache does not suggest that at all.

This should be reported upstream so documentation can be improved.
I'm a bit worried that this behaviour will break (where they might change how to mark it as a prefix).

Reply:
[Please see here for the docs](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/caching-dependencies-to-speed-up-workflows#matching-a-cache-key):
> When a key doesn't match directly, the action searches for keys prefixed with the restore key. If there are multiple partial matches for a restore key, the action returns the most recently created cache.

> I feel like this should be an issue on actions/cache
I agree. For now, this will have to do.

> I'm a bit worried that this behaviour will break.
I think this is intended to be a functional solution to the issue. It appears to be by-design how this works, but I think instead it would be better if they at least provided a way to just refresh the cache.